addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1096,18 +1096,31 @@ const Wysiwyg = Widget.extend({
                             backgroundGradient = backgroundImage;
                         }
                     }
+                    const hadSelection = range && (range.startContainer !== range.endContainer || range.startOffset !== range.endOffset);
                     colorpicker = new ColorPaletteWidget(this, {
                         excluded: ['transparent_grayscale'],
                         $editable: $(this.odooEditor.editable), // Our parent is the root widget, we can't retrieve the editable section from it...
                         selectedColor: backgroundGradient || $(targetElement).css(eventName === "foreColor" ? 'color' : 'backgroundColor'),
+                        selectedTab: backgroundGradient ? 'gradients' : 'theme-colors',
                         withGradients: true,
                     });
                     colorpicker.on('custom_color_picked color_picked', null, ev => {
+                        if (hadSelection) {
+                            this.odooEditor.resetLostSelection();
+                        }
                         this._processAndApplyColor(eventName, ev.data.color);
-                        this.odooEditor.historyStep();
+                        colorpicker.setSelectedColor(null, ev.data.color, false);
```

Comment:
That's strange:
- I receive an event from the colorpicker saying "red" is selected
- I do stuff with that red
- I say to the colorpicker "hey you sent me the color "red"" -> :thinking: 

Maybe something to call like "update toolbar" so that it rereads the selected DOM and update the UI accordingly.

Reply:
I added a replacement of this line as a wip commit... not sure we will want to keep it as it ultimately does only update colorpickers related to gradients.

Reply:
I finally found the problem I had: the colorpicker was triggering a preview event upon `setSeletedColor`.
The line replacement is now in a properly named commit, while the colorpicker fix is included in the fix commit that already touched `setSelectedColor`.